test/testdata/infer/dig.rb

```diff
@@ -0,0 +1,86 @@
+# typed: true
+# enable-suggest-unsafe: true
+
+extend T::Sig
+
+class MyClass; end
+
+sig {params(response: T::Hash[String, T::Hash[String, T::Array[T.noreturn]]]).void}
+def example1(response)
+  x = response.dig("", "", 0, 0)
+  #                           ^ error: Unused arguments to `dig`
+  T.reveal_type(x) # error: `NilClass`
+end
+
+sig {params(opts: T::Hash[String, T.any(Integer, T::Hash[String, String])]).void}
+def example2(opts)
+  x = opts.dig("data", "_id")
+  #                    ^^^^^ error: Method `dig` does not exist on `Integer` component of `T.any(T::Hash[String, String], Integer)`
+  T.reveal_type(x) # error: `T.untyped`
+end
+
+class MyClass
+  def my_method; end
+end
+
+sig {params(opts: T::Array[MyClass]).void}
+def example3(opts)
+  x = opts.dig(0)
+  T.reveal_type(x) # error: `T.nilable(MyClass)`
+  x.my_method
+  # ^^^^^^^^^ error: Method `my_method` does not exist on `NilClass` component of `T.nilable(MyClass)`
+end
+
+AShapeType = T.type_alias do
+  {
+    key: {
+      nested: T::Array[{x: Integer, y: Integer}]
+    },
+    another: Integer,
+  }
+end
+
+# Shapes are bad--ideally lot of these would be errors
+sig {params(opts: AShapeType).void}
+def example4(opts)
```

Comment:
The `TODO` in the corresponding exp file says we should have some tuple tests too.

Reply:
`example7`